src/main/java/com/thealgorithms/bitmanipulation/HighestSetBit.java

```diff
@@ -0,0 +1,30 @@
+package com.thealgorithms.bitmanipulation;
+import java.util.Optional;
+
+/**
+ * Find Highest Set Bit
+ * @author Bama Charan Chhandogi (https://github.com/BamaCharanChhandogi)
+ */
+
+public final class HighestSetBit {
+    private HighestSetBit() {
+    }
+
+    public final static Optional<Integer> findHighestSetBit(int num) {
+        if (num < 0) {
+            throw new IllegalArgumentException("Input cannot be negative");
+        }
+
+        if (num == 0) {
+            return Optional.empty();
+        }
+
+        int position = 0;
+        while (num > 0) {
+            num >>= 1;
+            position++;
+        }
+
+        return Optional.of(position - 1);
+    }
+}
```

Comment:
```suggestion
}

```

Does anyone know why `clang-format` is not complaining about that?